### PR TITLE
Upgrade Hibernate to 5.1.1

### PR DESCRIPTION
**Pull request by nguyenfilip**
This upgrade is needed to support EHCache. Reviewer please also verify this works with JDK 1.7 - I checked it, but just to be sure.

### CODE REVIEW

**Comment by nguyenfilip** [comment]
Investigation is needed to figure out why the Transactional annotations are necessary for tests to pass.

**Comment by nguyenfilip** [comment]
retest this please

**Comment by nguyenfilip** [comment]
I cannot get the unit tests to fail the same way again. Now everything looks happy even without Transactional annotations

**Comment by mstead** [comment]
Out of curiosity, what changed that now requires us to add the new @Transactional annotations?

File: server/src/main/java/org/candlepin/model/JobCurator.java [hunk 1]
```diff
@@ -59,6 +61,7 @@ public JobStatus cancel(String jobId) {
         return result;
     }
 
+    @Transactional
```
Comment (nguyenfilip) [inline server/src/main/java/org/candlepin/model/JobCurator.java]: @mstead I dont know exact answer, but the reason we need to add Transactional is that new Hibernate doesn't allow running update HQL outside of a transaction. If you think about it it is more appropriate behavior anyway.

**Comment by nguyenfilip** [comment]
To review with JDK 1.6. 
install Oracle Java 1.6
```
 cd into dir where you want your java to be installed 
 wget http://file.rdu.redhat.com/fnguyen/jdk-6u45-linux-x64.bin
 chmod +x jdk-6u45-linux-x64.bin
./jdk-6u45-linux-x64.bin
```
Modify JAVA_HOME for tomcat:
```
sudo vi /etc/tomcat/tomcat.conf
export JAVA_HOME=/home/fnguyen/down/jdk1.6.0_45
```

make sure you build with 1.6 as well - export JAVA_HOME before you run buildr or deploy scripts...

Before building candlepin, remove checks project (Swagger checkstyle plugin) 
```
rm -rf checks/
``
Deploy on MySQL (because Postgres JDBC driver requires Java 1.8 and working around that would  be hassle again)

**Comment by mstead** [comment]
Tested against 1.7 and 1.8.

**Pull request merged**